ra2ce/analyses/analysis_config_data/analysis_config_data.py

```diff
@@ -22,24 +22,137 @@
+    # general
+    weighing: str = ""  # should be enum
+    loss_per_distance: str = ""
+    loss_type: str = ""  # should be enum
+    disruption_per_category: str = ""
+    traffic_cols: str = ""  # should be list?
+    # losses
+    duration_event: float = math.nan
+    duration_disruption: float = math.nan
+    fraction_detour: float = math.nan
+    fraction_drivethrough: float = math.nan
+    rest_capacity: float = math.nan
+    maximum_jam: float = math.nan
+    partofday: str = ""
+    # accessiblity analyses
+    aggregate_wl: str = ""  # should be enum
+    threshold: float = math.nan
+    threshold_destinations: float = math.nan
+    uniform_duration: float = math.nan
+    gdp_percapita: float = math.nan
+    equity_weight: str = ""
+    calculate_route_without_disruption: bool = False
+    buffer_meters: float = math.nan
+    threshold_locations: float = math.nan
+    category_field_name: str = ""
+    save_traffic: bool = False
 
+
+@dataclass
+class AnalysisSectionDirect(AnalysisSectionBase):
+    # adaptation/effectiveness measures
+    return_period: float = math.nan
+    repair_costs: float = math.nan
+    evaluation_period: float = math.nan
+    interest_rate: float = math.nan
+    climate_factor: float = math.nan
+    climate_period: float = math.nan
+    # road damage
+    damage_curve: str = ""
+    event_type: str = ""
+    risk_calculation: str = ""
+    create_table: bool = False
+    file_name: Optional[Path] = None
+
+
+@dataclass
 class AnalysisConfigData(ConfigDataProtocol):
-    @classmethod
-    def from_dict(cls, dict_values: dict) -> AnalysisConfigData:
-        _new_analysis_ini_config_data = cls()
-        _new_analysis_ini_config_data.update(**dict_values)
-        return _new_analysis_ini_config_data
+    root_path: Optional[Path] = None
+    input_path: Optional[Path] = None
+    output_path: Optional[Path] = None
+    static_path: Optional[Path] = None
+    project: ProjectSection = field(default_factory=lambda: ProjectSection())
+    analyses: list[AnalysisSectionBase] = field(default_factory=list)
+    files: Optional[dict[str, Path]] = field(default_factory=dict)
+    origins_destinations: Optional[OriginsDestinationsSection] = field(
+        default_factory=lambda: OriginsDestinationsSection()
+    )
+    network: Optional[NetworkConfigData] = field(
+        default_factory=lambda: NetworkConfigData()
+    )
+    hazard_names: Optional[list[str]] = field(default_factory=list)
+
+    def to_dict(self) -> dict:
+        _dict = self.__dict__
+        _dict["project"] = self.project.__dict__
+        _dict["direct"] = [dv.__dict__ for dv in self.direct]
+        _dict["indirect"] = [dv.__dict__ for dv in self.indirect]
+        _dict["files"] = [dv.__dict__ for dv in self.files]
+        return _dict
+
+    @property
+    def direct(self):
+        return list(
+            analysis
+            for analysis in self.analyses
+            if analysis.analysis in DirectAnalysisNameList
+        )
```

Comment:
Should you not be check here for the type of item in the list?
```suggestion
        return list(
            analysis
            for analysis in self.analyses
            if isinstance(analysis, DirectAnalysisNameList)
        )
```
Which can be reduced with  `filter`:
```suggestion
        return list(filter(lambda x: isinstance(x, DirectAnalysisNameList), self.analyses))
```

Reply:
DirectAnalysisNameList is not a type, just a list of strings, so I don't think this will work.